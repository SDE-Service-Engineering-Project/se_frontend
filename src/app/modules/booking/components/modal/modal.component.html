<div class="modal-header">
  <h4 class="modal-title" id="modal-title">Cancel Booking</h4>
  <button
    type="button"
    class="btn-close"
    data-testid="modal-close-btn"
    aria-label="Close button"
    aria-describedby="modal-title"
    (click)="dismiss('canceled')"
  ></button>
</div>
<div class="modal-body">
  <p>Are you sure you want to cancel your Booking?</p>
  <p>
    <span class="text-danger"
      >This operation can not be undone!</span
    >
  </p>
</div>
<div class="modal-footer">
  <button
    type="button"
    class="btn btn-outline-secondary"
    data-testid="modal-cancel-btn"
    (click)="dismiss('canceled')"
  >
    Cancel
  </button>
  <button
    type="button"
    ngbAutofocus
    class="btn btn-danger"
    data-testid="modal-confirm-btn"
    (click)="close('confirmed')"
  >
    Ok
  </button>
</div>
