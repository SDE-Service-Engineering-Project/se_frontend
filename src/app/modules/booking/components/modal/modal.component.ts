import { Component } from '@angular/core';
import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';

@Component({
  selector: 'app-modal',
  templateUrl: './modal.component.html',
  styleUrls: ['./modal.component.sass'],
})
export class ModalComponent {
  constructor(private modal: NgbActiveModal) {}

  close(reason: string) {
    this.modal.close(reason);
  }

  dismiss(reason: string) {
    this.modal.dismiss(reason);
  }
}
