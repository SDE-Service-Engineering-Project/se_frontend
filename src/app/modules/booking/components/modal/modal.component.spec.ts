import { ComponentFixture, TestBed } from '@angular/core/testing';

import { ModalComponent } from './modal.component';
import {
  byTestId,
  createComponentFactory,
  Spectator,
} from '@ngneat/spectator/jest';
import { NgbActiveModal, NgbModal } from '@ng-bootstrap/ng-bootstrap';

describe('ModalComponent', () => {
  let component: ModalComponent;
  let fixture: ComponentFixture<ModalComponent>;
  let spectator: Spectator<ModalComponent>;
  let modal: NgbActiveModal;
  let modalSpy: jest.SpyInstance;

  const createComponent = createComponentFactory({
    component: ModalComponent,
    providers: [NgbActiveModal],
  });

  beforeEach(() => {
    spectator = createComponent();
    fixture = spectator.fixture;
    component = spectator.component;

    modal = spectator.inject(NgbActiveModal);
  });

  it('should render the component', () => {
    expect(fixture.debugElement).toMatchSnapshot();
  });

  it('should dismiss the modal on close button click with the reason canceled', () => {
    modalSpy = jest.spyOn(modal, 'dismiss');
    getCloseBtn().click();
    spectator.detectChanges();
    expect(modalSpy).toHaveBeenCalledWith('canceled');
  });

  it('should dismiss the modal on cancel button click with the reason canceled', () => {
    modalSpy = jest.spyOn(modal, 'dismiss');
    getCancelBtn().click();
    spectator.detectChanges();
    expect(modalSpy).toHaveBeenCalledWith('canceled');
  });
  it('should close the modal on confirm button click with the reason confirmed', () => {
    modalSpy = jest.spyOn(modal, 'close');
    getConfirmBtn().click();
    spectator.detectChanges();
    expect(modalSpy).toHaveBeenCalledWith('confirmed');
  });

  function getCloseBtn(): HTMLButtonElement {
    return spectator.query(byTestId('modal-close-btn')) as HTMLButtonElement;
  }
  function getCancelBtn(): HTMLButtonElement {
    return spectator.query(byTestId('modal-cancel-btn')) as HTMLButtonElement;
  }
  function getConfirmBtn(): HTMLButtonElement {
    return spectator.query(byTestId('modal-confirm-btn')) as HTMLButtonElement;
  }
});
