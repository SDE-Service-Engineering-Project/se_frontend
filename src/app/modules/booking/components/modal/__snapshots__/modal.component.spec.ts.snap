// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalComponent should render the component 1`] = `
DebugElement {
  "nativeNode": <div
    id="root0"
  >
    <div
      class="modal-header"
    >
      <h4
        class="modal-title"
        id="modal-title"
      >
        Cancel Booking
      </h4>
      <button
        aria-describedby="modal-title"
        aria-label="Close button"
        class="btn-close"
        data-testid="modal-close-btn"
        type="button"
      />
    </div>
    <div
      class="modal-body"
    >
      <p>
        Are you sure you want to cancel your Booking?
      </p>
      <p>
        <span
          class="text-danger"
        >
          This operation can not be undone!
        </span>
      </p>
    </div>
    <div
      class="modal-footer"
    >
      <button
        class="btn btn-outline-secondary"
        data-testid="modal-cancel-btn"
        type="button"
      >
         Cancel 
      </button>
      <button
        class="btn btn-danger"
        data-testid="modal-confirm-btn"
        ngbautofocus=""
        type="button"
      >
         Ok 
      </button>
    </div>
  </div>,
}
`;
